qcodes_contrib_drivers/drivers/Vaunix/LDA.py

```diff
@@ -0,0 +1,291 @@
+        """
+        Switch to this channel and set to ``value`` .
+        """
+        self._switch_channel()
+        value = round(value / self.scaling)
+        error_msg = self.dll_set_function(value)
+        if error_msg != 0:
+            raise RuntimeError(f'{self.dll_set_function.func.__name__} '
+                               f'returned error {error_msg}')
+
+
+class LdaAttenuation(LdaParameter):
+    """
+    Attenuation of one channel in the LDA.
+    """
+    scaling = 0.05  # integers returned by the API correspond to 0.05 dB
+
+    def __init__(self, name: str,
+                 instrument: Union[Vaunix_LDA, LdaChannel],
+                 **kwargs):
+        dll = instrument.root_instrument.dll
+        super().__init__(name, instrument,
+                         dll_get_function=dll.fnLDA_GetAttenuationHR,
+                         dll_set_function=dll.fnLDA_SetAttenuationHR,
+                         vals=Numbers(0, 63),  # replace below
+                         unit="dB",
+                         label="Attenuation",
+                         **kwargs,
+                         )
+
+        self._switch_channel()
+        min_att = dll.fnLDA_GetMinAttenuationHR(self.reference) * self.scaling
+        max_att = dll.fnLDA_GetMaxAttenuationHR(self.reference) * self.scaling
+        self.vals = Numbers(min_att, max_att)
+
+
+class LdaWorkingFrequency(LdaParameter):
+    """
+    Working frequency of one channel of the LDA. Not supported on all models.
+    """
+    scaling = 100_000  # integers returned by the API correspond to 100kHz
+
+    def __init__(self, name: str,
+                 instrument: Union[Vaunix_LDA, LdaChannel],
+                 **kwargs):
+        """
+        Attenuation of one channel in the LDA.
+
+        Args:
+            name: parameter name
+            instrument: parent instrument, either LDA or LDA channel
+        """
+        dll = instrument.root_instrument.dll
+        super().__init__(name, instrument,
+                         dll_get_function=dll.fnLDA_GetWorkingFrequency,
+                         dll_set_function=dll.fnLDA_SetWorkingFrequency,
+                         unit="Hz",
+                         label="Working frequency",
+                         docstring="Frequency at which the "
+                                   "attenuation is most accurate.",
+                         **kwargs
+                         )
+
+    @classmethod
+    def get_validator(cls, root_instrument: Vaunix_LDA) -> Optional[Numbers]:
+        """
+        Returns validator for working frequency, if ``root_instrument``
+        supports it. Else returns None.
+        """
+        max_freq = root_instrument.dll.fnLDA_GetMaxWorkingFrequency(
+                    root_instrument.reference) * cls.scaling
+        min_freq = root_instrument.dll.fnLDA_GetMinWorkingFrequency(
+                    root_instrument.reference) * cls.scaling
+        # if feature is not supported, these values will be equal
```

Comment:
shall then this be the condition instead of "whether `Numbers(min_freq, max_freq)` raises `TypeError` or not"?

Reply:
Fixed in 5d9a29e.